### ISSUE TITLE
Entity Ownership Changed notice is displaying when the module isn't shared with any roles.

### ISSUE THREAD

**Issue by techanvil** [google/site-kit-wp#6952]
## Bug Description

When changing the owned settings for the Analytics or Search Console module as a user who isn't the module owner, the "Entity Ownership Changed" notice is displayed regardless of whether the module is actually shared.

As per the [original AC](https://github.com/google/site-kit-wp/issues/4825), this notice should only be displayed when the module has been shared with one or more roles.

> * If `dashboardSharing` is enabled and the user has changed any of the owned settings (see #5121), and is not the owner themselves, **and the module is currently being shared with one or more roles:**  
>  display a new ⚠️ notice at the bottom of the edit view with the following content:  
> >  By clicking {submit button text}, you’re granting other users view-only access to data from {module name} via your Google account. You can always manage this later in the dashboard sharing settings.

## Steps to reproduce

1. Setup Site Kit, ensuring `dashboardSharing` is enabled. However do not share any modules.
2. Connect Analytics.
3. Login as a second admin and navigate to the Analytics settings.
4. Change one of the Analytics owned settings (e.g. the Property dropdown).
5. See the "Entity Ownership Changed" notice displayed as per the screenshot below.

Note the same applies for the Search Console module. In order to allow the Search Console Property dropdown to be changeable, follow the [QAB for #4825](https://github.com/google/site-kit-wp/issues/4825), specifically the point "To verify Search Console settings".

## Screenshots

![image.png](https://images.zenhubusercontent.com/6133937d41f397090b8344a7/cc133e17-f5ff-4928-b4c4-704b07798ba9)

## Additional Context

 - PHP Version: any
 - OS: any
 - Browser: any
 - Plugin Version: spotted 1.99.0
 - Device: any

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The "Entity Ownership Changed" notice, displayed for the Analytics and Search Console modules, should only be displayed when the module has been shared with one or more roles.

## Implementation Brief

* 

### Test Coverage

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
Thanks @techanvil – I believe this is a duplicate of #6633, no?

**Comment by techanvil** [comment]
Hey @aaemnnosttv, thanks for spotting that - I had looked to see if it was already covered but didn't spot that issue (and didn't remember commenting on it either, despite being only three weeks ago, due to the sheer amount of issues that I've looked at since then).

I think it's probably best to close this issue off in favour of #6633 in that case.

@jimmymadon I notice you've picked this up, but (duplicate nature aside) it's not an issue we need to be focusing on just yet as it's not a GA4 Reporting issue.
